Add Copilot agent onboarding instructions (#43)
cpholguera/frooky: .github/copilot-instructions.md

   - Outputs to `frooky/agent/dist/agent-{android,ios}.js`
   - **CRITICAL**: Agent artifacts MUST exist before Python package build

2. **Python Package Build**:
   ```bash
   python -m build
   ```
   - Packages the Python CLI and includes pre-built agent artifacts
   - Uses `setuptools-scm` for versioning from git tags
   - Outputs wheel (`.whl`) and source tarball (`.tar.gz`) to `dist/`

### Development Setup

To set up a local development environment:

```bash
# 1. Create and activate Python virtual environment
python3 -m venv venv
source venv/bin/activate

# 2. Compile Frida agents
./compileAgent.sh --dev

# 3. Install CLI in editable mode
pip install -e .

# 4. Verify installation
which frooky  # Should point to venv/bin/frooky
frooky --help
```

### Watch Mode for Active Development

For iterative agent development, use watch mode:

```bash
cd frooky/agent
npm run watch-android   # Auto-recompile on Android agent changes
npm run watch-ios       # Auto-recompile on iOS agent changes
```

Note: Use either `watch-android` or `watch-ios` depending on which platform you're working on.

## Testing & CI/CD

### Test Infrastructure

> - **No unit tests currently exist** - the project uses CI verification instead

Reviewer: ```suggestion
- **No unit tests currently exist** - the project only does CI verification at the moment, but testing of the codebase is in planning
```

   - Example: `git --no-pager status`, `git --no-pager diff`

## Common Tasks

### Adding New Agent Hooks
1. Edit TypeScript files in `frooky/agent/android/` or `frooky/agent/ios/`
2. Recompile: `cd frooky/agent && npm run dev-{android|ios}`
3. Test locally with `pip install -e .` and run `frooky` commands

### Modifying Python CLI
1. Edit `frooky/cli.py` or `frooky/frida_runner.py`
2. Changes are immediately available with `pip install -e .`
3. Test with `frooky --help` or relevant commands

### Adding Dependencies
- **Python**: Add to `dependencies` array in `pyproject.toml`
- **Node.js**: Run `cd frooky/agent && npm install --save-dev <package>`

### Documentation Updates
- Main docs are in `docs/` directory
- README.md provides quick start and examples
- Usage guide: `docs/usage.md`
- Development guide: `docs/develop.md`

## Key Files to Understand

### Python Side
- **`frooky/cli.py`**: Argument parsing, CLI entry point
- **`frooky/frida_runner.py`**: Core logic for loading hooks, attaching to processes, injecting agents
- **`pyproject.toml`**: Project metadata, dependencies, build configuration

### Agent Side

> - **`frooky/agent/build.js`**: Custom build orchestrator (handles TypeScript compilation, file watching)

Reviewer 2: ```suggestion
### Python Side

- **`frooky/cli.py`**: Argument parsing, CLI entry point
- **`frooky/frida_runner.py`**: Core logic for loading hooks, attaching to processes, injecting agents
- **`pyproject.toml`**: Project metadata, dependencies, build configuration

### Agent Side

- **`frooky/agent/build.js`**: Custom build orchestrator (handles TypeScript compilation, file watching)
```

- **`frooky/agent/android/`**: Android-specific hook implementations
- **`frooky/agent/ios/`**: iOS-specific hook implementations
- **`frooky/agent/package.json`**: Frida bridge dependencies, build scripts

## Workflow for Code Changes

1. **Identify scope**: Python CLI, Android agent, iOS agent, or docs?
2. **Set up dev environment**: Virtual env + compile agents
3. **Make changes**: Edit relevant files
4. **Rebuild as needed**:
   - Android agent changes: `cd frooky/agent && npm run dev-android`
   - iOS agent changes: `cd frooky/agent && npm run dev-ios`
   - Python changes: No rebuild needed with `pip install -e .`
5. **Test manually**: Run `frooky` commands against test apps/hooks
6. **Verify CI would pass**: Run full build + install verification locally
7. **Update docs** if user-facing behavior changes

## Platform-Specific Notes

### Android

> - Hooks Java/Kotlin methods using Frida's Java bridge

Reviewer: ```suggestion
- Hooks Java/Kotlin methods using Frida's Java bridge (`frida-java-bride`)
```
